platform/util/platform-util/src/main/java/org/codice/ddf/platform/util/properties/PropertiesLoader.java

```diff
@@ -108,108 +122,140 @@ public Properties loadProperties(String propertiesFile) {
+
+    properties = substituteSystemPropertyPlaceholders(properties);
+    return properties;
+  }
+
+  /** Default property loading strategy. */
+  @SuppressWarnings("squid:S1172" /* Used in bi-function */)
+  @VisibleForTesting
+  static Properties attemptLoadWithSpring(String propertiesFile, ClassLoader classLoader) {
+    Properties properties = new Properties();
+    try {
+      LOGGER.debug(
+          "Attempting to load properties from {} with Spring PropertiesLoaderUtils.",
+          propertiesFile);
+      properties = PropertiesLoaderUtils.loadAllProperties(propertiesFile);
+    } catch (IOException e) {
+      LOGGER.debug("Unable to load properties using default Spring properties loader.", e);
+    }
+    return properties;
+  }
+
+  /** Try loading properties using Spring and a provided class loader. */
+  @VisibleForTesting
+  static Properties attemptLoadWithSpringAndClassLoader(
+      String propertiesFile, ClassLoader classLoader) {
+    Properties properties = new Properties();
+    try {
+      LOGGER.debug(
+          "Attempting to load properties from {} with Spring PropertiesLoaderUtils with class loader.",
+          propertiesFile);
+      if (classLoader != null) {
+        properties = PropertiesLoaderUtils.loadAllProperties(propertiesFile, classLoader);
+      } else {
+        properties =
+            PropertiesLoaderUtils.loadAllProperties(
+                propertiesFile, PropertiesLoader.class.getClassLoader());
       }
+    } catch (IOException e) {
+      LOGGER.debug("Unable to load properties using default Spring properties loader.", e);
+    }
+    return properties;
+  }
 
-      if (error || properties.isEmpty()) {
-        LOGGER.debug("Attempting to load properties from file system: {}", propertiesFile);
-        File propFile = new File(propertiesFile);
-        // If properties file has fully-qualified absolute path (which
-        // the blueprint file specifies) then can load it directly.
-        if (propFile.isAbsolute()) {
-          LOGGER.debug("propertiesFile {} is absolute", propertiesFile);
-          propFile = new File(propertiesFile);
+  /**
+   * Try loading the properties directly from the file system. If the properties file has a
+   * fully-qualified absolute path (which is what the blueprint file should specify) then it can be
+   * loaded directly, using this method. Otherwise the path will be considered relative and attempts
+   * will be made with {@code karaf.home} and {@code ddf.home} property values prepended to the
+   * original path.
+   */
+  @SuppressWarnings("squid:S1172" /* Used in bi-function */)
+  @VisibleForTesting
+  static Properties attemptLoadWithFileSystem(String propertiesFile, ClassLoader classLoader) {
+    LOGGER.debug("Attempting to load properties from file system: {}", propertiesFile);
+    File propFile = new File(propertiesFile);
+    Properties properties = new Properties();
+
+    if (propFile.isAbsolute()) {
```

Comment:
❓  If the prop file path is absolute, shouldn't we be confirming that it is beneath `$ddf.home` before trying to read it?

Reply:
In fact, don't we already have a utility that can normalize a given input path?

Reply:
`AbsolutePathResolver`?

Reply:
We aren't supposed to change this classes functionality at all. Too many things depend on this class including integrators. Message @Lambeaux if you need a more in depth explanation.

Reply:
Would not change the functionality to use extant code to do what you're doing here. @Lambeaux would you please take a look and confirm that the `AbsolutePathResolver` would fit the needs here?